contracts/feirari/RewardsDistributorAdmin.sol

```diff
@@ -0,0 +1,129 @@
+pragma solidity ^0.8.0;
+
+import "../refs/CoreRef.sol";
+import "./IRewardsDistributorAdmin.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+contract RewardsDistributorAdmin is IRewardsDistributorAdmin, CoreRef, AccessControlEnumerable {
+
+    bytes32 public constant override AUTO_REWARDS_DISTRIBUTOR = keccak256("AUTO_REWARDS_DISTRIBUTOR");
+
+    /// @notice rewards distributor contract
+    IRewardsDistributorAdmin public rewardsDistributorContract;
+
+    /// @param coreAddress address of core contract
+    /// @param _rewardsDistributorContract admin rewards distributor contract
+    constructor(
+        address coreAddress,
+        IRewardsDistributorAdmin _rewardsDistributorContract,
+        address[] memory _autoRewardDistributors
+    ) CoreRef(coreAddress) {
+        rewardsDistributorContract = _rewardsDistributorContract;
+        /// @notice Governance should create new admin role for this contract
+        /// and then grant this role to all AutoRewardsDistributors so that they can call this contract
+        _setContractAdminRole(keccak256("TRIBAL_CHIEF_ADMIN_ROLE"));
+        /// assign the admin role of AUTO_REWARDS_DISTRIBUTOR to DEFAULT_ADMIN_ROLE
+        _setRoleAdmin(AUTO_REWARDS_DISTRIBUTOR, DEFAULT_ADMIN_ROLE);
+
+        /// give all AutoRewardsDistributor contracts the proper role so that they can set borrow and supply speeds
+        for (uint256 i = 0; i < _autoRewardDistributors.length; i++) {
+            _setupRole(AUTO_REWARDS_DISTRIBUTOR, _autoRewardDistributors[i]);
+        }
+    }
+
+    /**
+      * @notice Begins transfer of admin rights. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @dev Admin function to begin change of admin. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @param newPendingAdmin New pending admin.
+      */
+    function _setPendingAdmin(address newPendingAdmin) external override onlyGovernor {
+        rewardsDistributorContract._setPendingAdmin(newPendingAdmin);
+    }
+
+    /**
+      * @notice Accepts transfer of admin rights. msg.sender must be pendingAdmin
```

Comment:
this comment is slightly inaccurate. "this" must be pendingAdmin not the sender

Reply:
I just copy and pasted this comment from the rari codebase as it is a passthrough.